src/book.py

```diff
@@ -176,6 +178,79 @@ def _read_coinbase(self, file_path: Path) -> None:
                     self.append_operation("Fee", utc_time, platform,
                                           eur_fee, "EUR", row, file_path)
 
+    def _read_kraken_trades(self, file_path: Path) -> None:
+        log.error(f"{file_path.name}: "
+                  f"Looks like this is a Kraken 'Trades' history, but we need the 'Ledgers' history. "
+                  f"(See: https://github.com/provinzio/CoinTaxman/wiki/Exchange:-Kraken)")
+
+    def _read_kraken_ledgers(self, file_path: Path) -> None:
+        platform = "kraken"
+        operation_mapping = {
+            "trade": None,  # Buy or Sell order
+            "spend": "Sell",  # Sell ordered via 'Buy Crypto' button
+            "receive": "Buy",  # Buy ordered via 'Buy Crypto' button
+            "transfer": "Airdrop",
+            "reward": "StakingInterest",
+            "deposit": "Deposit",
+            "withdrawal": "Withdraw",
+        }
+
+        # Need to track state of "duplicate entries" for deposits / withdrawals:
+        #   dup_state["deposit"] == 1: Deposit is broadcast to blockchain            <-- Taxable event (is in public trade history)
+        #   dup_state["deposit"] == 2: Deposit is credited to Kraken account         <-- Skipped
+        #   dup_state["withdrawal"] == 1: Withdrawal is requested in Kraken account  <-- Skipped
+        #   dup_state["withdrawal"] == 2: Withdrawal is broadcast to blockchain      <-- Taxable event (is in public trade history)
+        dup_state, dup_skip = {"deposit": 0, "withdrawal": 0}, {"deposit": 2, "withdrawal": 1}
+        # See: https://support.kraken.com/hc/en-us/articles/360001169443-Why-there-are-duplicate-entries-for-deposits-withdrawals
```

Comment:
> Should it be "deposit": 1? Your calculations below never result in 2.

I guess I was lazy coding this. I'm comparing _after_ incrementing, thus saving a line or two of code, but instead "wrapping around" the value to compare with. I'll now make this very explicit, without using modulo.

Reply:
EDIT²: On a second thought, I implemented it like this in order to be able to select which event is the taxable one. I'm absolutely no expert on this. If you can confirm that the "broadcast to blockchain" event is the right one, I can hard-code this. Otherwise, lets leave it like that, using counters?!

Reply:
I believe that the broadcast to blockchain is the relevant event. The other seems to me internal only.

Reply:
Ok, I thought so, too, but wasn't 100% sure about it.

But since we need to skip the _first_ deposit event, and need to skip the _second_ withdrawal event, I think using counters is still the most simple solution?

Reply:
Well, I still like the boolean idea, which might be easier to understand? The current solution looks too complicated for the use case.

If you want to stand with your current implementation, you could change `(dup_skip[_type] + 1) % 2` to just `dup_skip[_type]` and change the skipping values to 0 and 1. Which removes one layer of complexity.